apps/admin/backend/src/util/adjudication.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  BallotPageLayout,
+  BallotStyleId,
+  ContestId,
+  Id,
+} from '@votingworks/types';
+import { loadImageData, toDataUrl } from '@votingworks/image-utils';
+import { Store } from '../store';
+import { BallotImageView } from '../types';
+import { rootDebug } from './debug';
+
+const debug = rootDebug.extend('adjudication');
+
+/**
+ * Determines the side of the ballot a contest is on.
+ */
+export function getContestSideFromLayout(
```

Comment:
This file was renamed from `.../write-in.ts` to `.../adjudication.ts`

`getContestSideFromLayout` is a new function. `getBallotImageView` is renamed from `getWriteInImageView`, because we now need ballot images for ballots without write-ins (any contest option can have a marginal mark). It is also slightly refactored as we now provide the `contestOption layouts` instead of the `writeInCoordinates` to enable zooming in on any contest option.